src/AutoConstructor.Generator/AutoConstructorGenerator.cs

```diff
@@ -228,41 +232,90 @@ private static string GenerateAutoConstructor(MainNamedTypeSymbolInfo symbol, Eq
 
         // Write type name line.
         writer.WriteNamedTypeSymbolInfoLine(symbol);
+        bool generateCtorWithParameters = fields.Any();
 
-        // Write type content.
         using (writer.WriteBlock())
         {
-            // Get all constuctor parameters from fields. 
-            FieldInfo[] constructorParameters = fields
-                .GroupBy(x => x.ParameterName)
-                .Select(x => x.Any(c => c.Type is not null) ? x.First(c => c.Type is not null) : x.First())
-                .ToArray();
-
-            // Write constructor documentation if enable.
-            if (generateConstructorDocumentation)
+            // Write constructor(s) themselves
+            if (generateCtorWithParameters)
+            {
+                WriteConstructorDeclaration(symbol, fields, options, false, writer);
+            }
+            if (symbol.AddParameterless)
             {
-                writer.WriteLine("/// <summary>");
-                writer.WriteLine($"/// {string.Format(CultureInfo.InvariantCulture, constructorDocumentationComment, symbol.Name)}");
-                writer.WriteLine("/// </summary>");
-                foreach (FieldInfo parameter in constructorParameters)
+                //add separation line between constructors
+                if (generateCtorWithParameters)
                 {
-                    writer.WriteLine($"/// <param name=\"{parameter.ParameterName}\">{parameter.Comment ?? parameter.ParameterName}</param>");
+                    writer.WriteLine("");
                 }
+                WriteConstructorDeclaration(symbol, new EquatableArray<FieldInfo>(), options, true, writer);
             }
+        }
+
+        // Close nested types.
+        foreach (NamedTypeSymbolInfo containingType in symbol.ContainingTypes)
+        {
+            writer.EndBlock();
+        }
+
+        // Close namespace.
+        if (!symbol.Namespace.IsGlobal)
+        {
+            writer.EndBlock();
+        }
+
+        return writer.ToString();
+    }
+
+    private static void WriteConstructorDeclaration(MainNamedTypeSymbolInfo symbol, EquatableArray<FieldInfo> fields, Options options, bool isParameterLess, IndentedTextWriter writer)
```

Comment:
I think there wasn't a nice way to add a second constructor generation easily, because the whole code generation was one big method - about 140 lines of code. I could leave it as is and call AddParameterless somewhere inside, but then we would have massive code duplication. I think it would be even better to split WriteConstructorDeclaration into more smaller methods, but that would be even more refactoring.

Reply:
I don't think this would have generated that much code duplication, and I think that having a method with multiple `isParameterless` checks is worse. I won't ask you to redo it all, but it had to be said.